fontforge/sfd.c

```diff
@@ -668,13 +669,21 @@ static void SFDDumpHintMask(FILE *sfd,HintMask *hintmask) {
     }
 }
 
-static void SFDDumpSplineSet(FILE *sfd,SplineSet *spl) {
+static void SFDDumpSplineSet(FILE *sfd, SplineSet *spl, int want_order2) {
     SplinePoint *first, *sp;
     int order2 = spl->first->next==NULL || spl->first->next->order2;
+    int reduce = (want_order2 && !order2);
+    if (order2 && !want_order2) IError("Asked for cubic when had quadratic");
+    SplineSet *nspl;
 
     for ( ; spl!=NULL; spl=spl->next ) {
+	if (reduce) {
+	    nspl = SSttfApprox(spl);
```

Comment:
OK, just one more thing: set `order2` to true here so that the methods that use it below get the right value. After that @frank-trampe can take another glance at the new changes, but it basically looks good to me.

Reply:
good idea

Reply:
Resolved in https://github.com/fontforge/fontforge/pull/3679/commits/8e2198411b453a5426faec0cd9d60f4a129a78fb